exporter/uptraceexporter/exporter.go

```diff
@@ -0,0 +1,239 @@
+// Copyright 2021 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package uptraceexporter
+
+import (
+	"context"
+	"encoding/binary"
+	"time"
+
+	"github.com/uptrace/uptrace-go/spanexp"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/label"
+	"go.opentelemetry.io/otel/metric"
+	"go.opentelemetry.io/otel/unit"
+	"go.uber.org/zap"
+)
+
+type traceExporter struct {
+	cfg    *Config
+	logger *zap.Logger
+	upexp  *spanexp.Exporter
+
+	backendLatency metric.Int64ValueRecorder
+	meterErr       error
+}
+
+func newTraceExporter(cfg *Config, logger *zap.Logger) (*traceExporter, error) {
+	upexp, err := spanexp.NewExporter(&spanexp.Config{
+		DSN: cfg.DSN,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	meter := otel.Meter(
+		"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/uptraceexporter")
```

Comment:
Do I really need to use such long name? :confused:

Reply:
I don't think we need. I wasn't aware that we should be using the OpenTelemetry API for this already, but for the legacy OpenCensus one this wasn't necessary:

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9f137ea75d249631c411b9d845e83a179032a5b8/exporter/loadbalancingexporter/metrics.go#L24

And then:
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/9f137ea75d249631c411b9d845e83a179032a5b8/exporter/loadbalancingexporter/resolver_dns.go#L125

Where did you see this usage?

Reply:
I try to follow docs at https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/metrics/api.md#obtaining-a-meter which mentions "io.opentelemetry.contrib.mongodb" for Java.

>I wasn't aware that we should be using the OpenTelemetry API for this already

Yeah, I guess metrics are not stable yet. But using OpenCensus makes even less sense (to me)...

What do you think if I just drop the metrics instrumentation for now? Add add them in a separate PR later when Otel Metrics are ready?

Reply:
I know your feeling, I had it too. But it's either that, or you won't have metrics for your component, which is even worse for you  and for your users. Besides, all other components are instrumented with OpenCensus at the moment and will likely be mass-migrated once we are ready with OpenTelemetry Metrics.

Reply:
>which is even worse for you and for your users

But I have tracing :) SendSpans is instrumented with Otel traces so I have even more data than with metrics. Besides Uptrace does not support metrics at the moment...

>are instrumented with OpenCensus

Just to make it clear  - should I replace Otel Metrics with OpenCensus for now?

Reply:
> Besides Uptrace does not support metrics at the moment

This is about understanding what the collector is doing. If a user of yours is using the collector and see no data in your service, how do you intend to figure out what's going on? The collector exposes a `/metrics` by default, which includes data generated by the individual components. If you think you have enough information elsewhere to support your users, that's fine, you can skip this.

> Just to make it clear - should I replace Otel Metrics with OpenCensus for now?

If you think you should have metrics to understand what your exporter is doing, yes. If you decide to not have metrics at all, I'm also fine.

Reply:
If this is not a hard requirement I would prefer to add Otel metrics later in a separate PR. Few other exporters I've checked don't have metrics too :)

I don't have experience with OpenCensus and learning it at this stage does not look worth it even if basic concepts are the same...

Reply:
Alright, feel free to update the PR to remove the metric.